### ISSUE TITLE
kubernetes-config loads literals even when turned off

### ISSUE THREAD

**Issue by SIMULATAN** [quarkusio/quarkus#38846]
### Describe the bug

Since the deployment in the generated manifest loads secrets like this:
```yml
envFrom:
  - configMapRef:
    name: postgres-config
```
literals used as environment variables (e.g. `${POSTGRES_USER}`) work despite [`quarkus.kubernetes-config.enabled`](https://quarkus.io/guides/kubernetes-config#quarkus-kubernetes-config_quarkus-kubernetes-config-enabled) being set to false. However, config maps featuring a `application.properties` do not work (as expected).

### Expected behavior

Either all loading works or none.
Considering the setting: no `configMapRef` if `kubernetes-config.enabled=false`

### Actual behavior

Environment variables are accessible in the config

### How to Reproduce?

1. Install `kubernetes-config` extension
2. set `quarkus.kubernetes-config.config-maps=literal-config,application-config`
3. deploy ConfigMaps to your kubernetes cluster
	<details><summary>Resources</summary>
	<p>

	```yml
	apiVersion: v1
	kind: ConfigMap
	metadata:
	  name: literal-config
	data:
	  MY_VALUE: hello
	---
	apiVersion: v1
	kind: ConfigMap
	metadata:
	  name: application-config
	data:
	  application.properties: |
	    quarkus.log.level=debug
	    my.property=hello
	```

	</p>
	</details>
4. use `${MY_VALUE}` somewhere in a config property
5. observe as `MY_VALUE` is set but debug logging is disabled and `my.property` doesn't exist

### Quarkus version or git rev

3.6.4

### Additional information

This is quite hard to spot as half of it works while the other half doesn't. In my case, I have two config maps: `postgres-config` and `backend-config`. The former contains literals (`POSTGRES_USER` & `POSTGRES_DB`) while the latter has a `application.properties` file. Since the former worked (and the 3 secrets loaded did too, including two that configure the application using the properties way!) I really didn't expect the config extension to be turned off like that.

**Comment by geoand** [comment]
Can share the Kubernetes manifest that is being used to deploy the application?

**Comment by SIMULATAN** [comment]
Well, after further debugging, it appears this is not caused by `kubernetes-config` but rather [`quarkus.kubernetes.env`](https://quarkus.io/guides/deploying-to-kubernetes#environment-variables-from-configmap).
> To add all key/value pairs from ConfigMap as environment variables just apply the following configuration, separating each ConfigMap to be used as source by a comma (,):
> `quarkus.kubernetes.env.configmaps=my-config-map,another-config-map`

I stumbled upon this configuration option when I first set the k8s integration up and used it as it seemed like it accomplishes the same thing. I only saw the Configuration Reference on it, which says `The optional list of ConfigMap names to load environment variables from`. Seeing as I started with only the postgres environment variables, I figured I'd use it.

Sorry, my bad!
